### ISSUE TITLE
Shapely install

### ISSUE THREAD

**Issue by darribas** [carsonfarmer/python_geospatial#3]
Trying to work my way through the tutorial. Stuck at the install part. I've created the environment as suggested in `install.md` and it seems to work just fine until `shapely`. `conda` seems to install it correctly:

``` python
In [1]: import shapely

In [2]:
```

However, when I try the following, I the an OSError:

``` python
In [2]: import shapely.geos
---------------------------------------------------------------------------
OSError                                   Traceback (most recent call last)
<ipython-input-2-298ab485fd08> in <module>()
----> 1 import shapely.geos

/Users/dani/anaconda/envs/scipygis/lib/python2.7/site-packages/shapely/geos.py in <module>()
     63     alt_paths = [os.path.join(sys.prefix, 'lib', 'libgeos_c.dylib')]
     64     _lgeos = load_dll('geos_c', fallbacks=alt_paths)
---> 65     free = load_dll('c').free
     66     free.argtypes = [c_void_p]
     67     free.restype = None

/Users/dani/anaconda/envs/scipygis/lib/python2.7/site-packages/shapely/geos.py in load_dll(libname, fallbacks)
     51     raise OSError(
     52         "Could not find library %s or load any of its variants %s" % (
---> 53             libname, fallbacks or []))
     54
     55

OSError: Could not find library c or load any of its variants []

In [3]:
```

Any idea on why this happens and, particularly, how I can fix it? All comments and help mostly appreciated. I'm running conda on Mavericks:

``` bash
(scipygis)kokopelli:Desktop dani$ conda info
Current conda install:

             platform : osx-64
        conda version : 3.5.5
       python version : 2.7.8.final.0
     root environment : /Users/dani/anaconda  (writable)
  default environment : /Users/dani/anaconda/envs/scipygis
     envs directories : /Users/dani/anaconda/envs
        package cache : /Users/dani/anaconda/pkgs
         channel URLs : http://repo.continuum.io/pkgs/free/osx-64/
                        http://repo.continuum.io/pkgs/pro/osx-64/
          config file : None
    is foreign system : False

(scipygis)kokopelli:Desktop dani$
```

**Comment by carsonfarmer** [comment]
Hmm, not quite sure what would be causing this? I assume you have developer tools installed on your system? What about if you `conda remove geos` and try `brew install`ing it instead, then `pip install`ing shapely that way? You don't get the self-contained install that `conda install` affords, but it should work (this is usually what I do)?

**Comment by darribas** [comment]
Nope :-( I've found Kelsey's trick [here](https://github.com/kjordahl/SciPy2013#installation-instructions) and that allows me to `import shapely.geos` without complain. However, when I install `cartopy` and try one of their [toy examples](http://scitools.org.uk/cartopy/docs/latest/matplotlib/intro.html):

``` python
In [3]: import cartopy.crs as ccrs

In [4]: import matplotlib.pyplot as plt

In [5]:

In [5]: ax = plt.axes(projection=ccrs.PlateCarree())
Assertion failed: (0 != cs), function GEOSCoordSeq_getSize_r, file geos_ts_c.cpp, line 3612.
Abort trap: 6
(scipygis)kokopelli:~ dani$
```

And throws me out. It looks like this is a `cartopy` issue, not a shapely, but my hunch is that it all goes back to my GEOS installation. I've tried a couple of combinations (conda or brew for geos with cartopy from pip or conda), and I always get that issue.

Any comment or help most welcome, but I understand this might be my own crappy setup full of different installations and layers at this point.

**Comment by carsonfarmer** [comment]
Ah ok, yes I think this is a GEOS installation issue. I think I ran into the same problem at one point. Unfortunately, it looks like `conda` is using a very recent version of GEOS, whereas `cartopy` is expecting an older one (or vice versa). I've found that _not_ using the `conda` versions and sticking with _all_ `pip` versions works best for my system.

**Comment by darribas** [comment]
That did the trick... Wohooooooooo!!!

To recap, I removed anything dealing with `conda` that depends on `geos`:

``` bash
conda remove geos, shapely, cartopy
```

Then installed geos with brew:

``` bash
brew install geos
```

Added Kelsey's workaround to my `.bash_profile`:

``` bash
export DYLD_FALLBACK_LIBRARY_PATH=$(HOME)/lib:/usr/local/lib:/lib:/usr/lib
```

And then `pip`'ed the two libraries:

``` bash
pip install shapely
pip install git+git://github.com/SciTools/cartopy.git
```

Doing that in both the general setup and/or the `scipygis` virtual env seems to get cartopy to finally work. Closing the issue on this. Thank you very much for the advice!
